src/_blog/webtransport.md

```diff
@@ -0,0 +1,252 @@
+Therefore, for the longest time, the only way to connect a libp2p client running in the browser to the rest of the network was using the [somewhat dated WebSocket protocol](https://www.rfc-editor.org/rfc/rfc6455).
+
+WebSocket works by first establishing a HTTP(S) connection (most commonly HTTP 1.1) to the server, and then performing a single HTTP request to “upgrade” the connection to a bidirectional byte-stream. In libp2p, this allowed us to then treat this WebSocket connection as a raw TCP connection.
+We run an additional libp2p handshake on that connection and apply a stream multiplexer ([yamux](https://github.com/libp2p/go-yamux) or [mplex](https://github.com/libp2p/specs/tree/master/mplex)). The added step is necessary because WebSocket is susceptible to a problem called [head-of-line blocking](https://hpbn.co/websocket/#websocket-multiplexing-and-head-of-line-blocking). Because the protocol doesn't support multiplexing natively and this can cause one message sent via WebSocket to block other messages behind it (even if the messages are independent of each other.)
+
+This has multiple drawbacks:
+
+- Slow time to connect
+    - It takes 6 network roundtrips until the libp2p connection is finally established because of the steps involved to establish a WebSocket connection.
+- Inefficiency
+    - We’re double-encrypting the data: The first time, it’s encrypted on the outer (HTTPS) connection, and then again by the libp2p security protocol.
+- Increased latency
+    - There's no native stream multiplexing in WebSockets, and even after we add our own multiplexer, each internal stream can still suffer from head-of-line blocking.
+
+This means that we never could expect the WebSocket transport to be a fast and performant transport to begin with. Given that WebSocket was the only connectivity option for browsers, we had no choice but to pay the performance penalty.
+
+In practice, a different obstacle prevented WebSocket from achieving widespread deployment in libp2p. When a browser connects to a website, in practically all cases it does so via [HTTPS](https://www.cloudflare.com/learning/ssl/what-is-https/). HTTPS enforces the use of *Secure WebSocket*, which is just a fancy name WebSocket over HTTPS (and not HTTP). This means that the server needs a valid [TLS certificate](https://aws.amazon.com/what-is/ssl-certificate/), i.e. a certificate signed by a [Certificate Authority](https://en.wikipedia.org/wiki/Certificate_authority) like [Let’s Encrypt](https://letsencrypt.org/).
+
+However, most libp2p nodes don’t have such a certificate. This is because libp2p nodes constitue a peer-to-peer decentralized network where participants can run nodes on home laptop or browsers as well as join or leave the network at will. Most nodes don’t even possess a domain name, which is a requirement for getting a certificate from many CAs. While it’s not too hard to obtain a TLS certificate, it’s non-trivial to do so fully automatically in a decentralized manner.
```

Comment:
```suggestion
However, most libp2p nodes don’t have such a certificate. This is because libp2p nodes constitue a peer-to-peer decentralized network where participants can run nodes on home laptop or browsers as well as join or leave the network at will. Most nodes don’t even possess a domain name, which is a requirement for getting a certificate from many CAs. While it’s not too hard to obtain a TLS certificate, it’s non-trivial to do so automatically in a decentralized manner.
```